Lib/test/test_capi/test_bytes.py

```diff
@@ -247,6 +247,39 @@ def test_resize(self):
         # CRASHES resize(NULL, 0, False)
         # CRASHES resize(NULL, 3, False)
 
+    def test_join(self):
+        """Test PyBytes_Join()"""
+        bytes_join = _testcapi.bytes_join
+
+        self.assertEqual(bytes_join(b'', []), b'')
+        self.assertEqual(bytes_join(b'sep', []), b'')
+
+        self.assertEqual(bytes_join(NULL, [b'a', b'b', b'c']), b'abc')
+        self.assertEqual(bytes_join(b'', [b'a', b'b', b'c']), b'abc')
+        self.assertEqual(bytes_join(b'-', [b'a', b'b', b'c']), b'a-b-c')
+        self.assertEqual(bytes_join(b' - ', [b'a', b'b', b'c']), b'a - b - c')
+        self.assertEqual(bytes_join(b'-', [bytearray(b'abc'),
+                                           memoryview(b'def')]),
+                         b'abc-def')
+
+        self.assertEqual(bytes_join(b'-', iter([b'a', b'b', b'c'])), b'a-b-c')
+
+        # invalid 'sep' argument type
+        with self.assertRaises(TypeError):
+            bytes_join(bytearray('sep'), [])
+        with self.assertRaises(TypeError):
+            bytes_join('unicode', [])
+        with self.assertRaises(TypeError):
+            bytes_join(123, [])
+
+        # invalid 'iterable' argument type
+        with self.assertRaises(TypeError):
+            bytes_join(b'', [b'bytes', 'unicode'])
+        with self.assertRaises(TypeError):
+            bytes_join(b'', [b'bytes', 123])
+
+        # CRASHES bytes_join(sep, NULL)
```

Comment:
I think that you will get a SystemError if remove additional assert and add NULLABLE().